### PR TITLE
Add generators and runtime configuration packages

### PR DESCRIPTION
**Pull request by jacobperron**
Moved (and renamed) from rosidl_defaults.

Related PR: https://github.com/ros2/rosidl_defaults/pull/22

### CODE REVIEW

File: rosidl_core_runtime/QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+This document is a declaration of software quality for the `rosidl_core_runtime` package, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).
```
Comment (ivanpauno) [inline rosidl_core_runtime/QUALITY_DECLARATION.md:1]: should there be a quality declaration for `rosidl_code_generators` as well?
could there only be one in the top level?

File: rosidl_core_runtime/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# rosidl_core_runtime
```
Comment (ivanpauno) [inline rosidl_core_runtime/README.md:1]: a top level readme and license file would be nice as well

**Review by ivanpauno — APPROVED** [review]
LGTM with minor comments addressed and green CI

File: rosidl_core_runtime/QUALITY_DECLARATION.md (hunk 1 above)
Comment (jacobperron) [reply rosidl_core_runtime/QUALITY_DECLARATION.md:1]: I'm not sure. There was _not_ a quality declaration in `rosidl_defaults`, so I did not create one here. It seems we only did quality declarations for the runtime packages.

Comment (ivanpauno) [reply rosidl_core_runtime/QUALITY_DECLARATION.md:1]: Okay, that makes sense to me

File: rosidl_core_runtime/README.md (hunk 2 above)
Comment (jacobperron) [reply rosidl_core_runtime/README.md:1]: There is already a top-level license file (I initialized the repository with it).

I've added/update READMEs now: [1d9ac52](https://github.com/ros2/rosidl_core/pull/1/commits/1d9ac528029211bc9dc8abb954cfbf09519ee67f)

Comment (ivanpauno) [reply rosidl_core_runtime/README.md:1]: > There is already a top-level license file (I initialized the repository with it).

Sorry, didn't see that.

> I've added/update READMEs now: [1d9ac52](https://github.com/ros2/rosidl_core/pull/1/commits/1d9ac528029211bc9dc8abb954cfbf09519ee67f)

Thanks!

**Review by ivanpauno — APPROVED** [review]

**Pull request merged**